plugins/SnapOnMove/src/main/java/org/micromanager/plugins/snaponmove/MainController.java

```diff
@@ -235,6 +234,12 @@ private void monitorLoop() {
             }
             doSnap();
             waitForChange();
+            synchronized (pauseLock_) {
+               while (pausedForAcquisition_) {
+                  pauseLock_.wait();
+                  waitForChange();
```

Comment:
It is better to avoid calling `waitForChange()` (which blocks for arbitrary duration) while holding `pauseLock_`, because that prevents another thread from clearing `pausedForAcquisition_`.
Best to set a boolean flag `shouldWaitAgain` inside the `synchronized` and call `waitForChange()` after exiting the `synchronized` when the flag is set.
Alternatively, it's probably okay to just leave out the `waitForChange()` here and let the next snap happen after an acquisition finished (though it would be good to test that this doesn't cause issues).